src/__tests__/mockServer.ts

```diff
@@ -42,7 +28,7 @@ export function createMockServer<T extends IncomingEvent>(
       )!;
       res.statusCode = result.statusCode!;
       Object.entries(result.headers ?? {}).forEach(([key, value]) => {
-        res.setHeader(key, value.toString());
+        res.setHeader(key, (value as string).toString());
```

Comment:
Before this was `unknown`, it was `string | number | boolean`, I think we should cast to that or cast as `any` with a comment. Casting to `string` and calling `toString()` on the thing smells a little funny.

Reply:
Updated to the `String(value)` runtime type as it accepts `any` (and therefore `unknown`) directly without manual coercion.